docs/hive-integration/kube-api-webhook.md

```diff
@@ -0,0 +1,49 @@
+# Webhook validation
+
+Admission webhooks are HTTP callbacks that receive admission requests and do something with them. There are two types of admission webhooks: validating admission webhook and mutating admission webhook.
+
+- Mutating Admission Webhook: These can mutate the object while it’s being created or updated, before it gets stored. It can be used to default fields in a resource requests, e.g. fields in Deployment that are not specified by the user.
+
+- Validating Admission Webhook: These can validate the object while it’s being created or updated, before it gets stored. It allows more complex validation than pure schema-based validation. e.g. cross-field validation.
+
+Read more about admission webhooks [here](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+
+Webhook validation enable adding validations based on the state of the CRDs.
+For example, changing the Spec of AgentClusterInstall or Agent after the installation started is not permitted.
```

Comment:
I think we can add some info about why I would run this webhook. From conversations I gathered that we are using this validating admission web hook to validate the CRDs before starting the installation. This isn't really clear here. Also I would remove the mutating webhook unless there are plans to add one (to fix configuration, maybe)? It could be confusing if our webhook isn't a mutating one and we mention that.

Reply:
Done